ReSharper.FSharp/src/FSharp.Psi.Features/src/Parsing/FSharpParser.fs

```diff
@@ -62,10 +62,12 @@ type FSharpParser(lexer: ILexer, document: IDocument, path: VirtualFileSystemPat
         let path = if isNotNull sourceFile then sourceFile.GetLocation() else null
         FSharpParser(lexer, document, path, sourceFile, checkerService, symbolsCache)
 
-    new (lexer, document, sourceFile, checkerService, symbolsCache) =
-        FSharpParser(lexer, document, FSharpParser.SandBoxPath, sourceFile, checkerService, symbolsCache)
+    new (lexer, document, sourceFile: IPsiSourceFile, checkerService, symbolsCache) =
+        let path = if sourceFile.Name.EndsWith(".fsi") then FSharpParser.SandBoxSignaturePath else FSharpParser.SandBoxPath
```

Comment:
`sourceFile` may be `null` here, it's used as a context in which a new node is being parsed, and we can just use its path.
```suggestion
        let path = if isNotNull sourceFile then sourceFile.GetLocation() else FSharpParser.SandBoxPath
```